python/nav/snmptrapd/trap.py

```diff
@@ -65,8 +65,8 @@ def _lookup_agent(self):
         conn = getConnection('snmptrapd')
         cur = conn.cursor()
         cur.execute(
-            "SELECT netboxid, sysname, roomid FROM netbox WHERE ip = %s",
-            (self.agent,),
+            "SELECT netboxid, sysname, roomid FROM gwportprefix JOIN interface USING (interfaceid) JOIN netbox USING (netboxid) WHERE ip = %s OR gwip = %s",
+            (self.agent, self.agent),
```

Comment:
Yes, this should pretty much do the trick :)

That is, this works, but could still be improved:

1. The SQL statement will now potentially return multiple rows (although the code picks only the first hit)
2. The line is longer than our convential max of 88 characters, but Black will not automatically wrap strings, so it needs to be done manually for clarity.
3. No need to specify the IP address argument twice.

I suggest something like this:

```suggestion
            "SELECT DISTINCT netboxid, sysname, roomid "
            "FROM gwportprefix "
            "JOIN interface USING (interfaceid) "
            "JOIN netbox USING (netboxid) "
            "WHERE %s IN (ip, gwip) "
            (self.agent, ),
```